src/robot/wifi_information.cpp

```diff
@@ -0,0 +1,138 @@
+#include "wifi_information.h"
+
+namespace Valeronoi::robot {
+
+class WifiInformationData : public QSharedData
+{
+public:
+    WifiInformationData() {
+        m_bssid = "00:00:00:00:00:00";
+        m_ssid = "Unknown Wifi";
+        m_signal = 0.0;
+    }
+
+    WifiInformationData(const WifiInformationData &other)
+        : QSharedData(other),
+        m_ssid(other.m_ssid),
+        m_bssid(other.m_bssid),
+        m_signal(other.m_signal)
+    { }
+
+    ~WifiInformationData() { }
+
+    QString m_ssid;
+    QString m_bssid;
+
+    double m_signal; // measurement
+
+//    // Optionally this could be expanded
+//    QString m_frequency;
+//    QString m_ip;
+//    QString m_ipv6;
+//    double m_upspeed; // Measurement
+};
+
+WifiInformation::WifiInformation() : m_data(new WifiInformationData)
+{
+
+}
+
+WifiInformation::WifiInformation(const QJsonObject &jsonObj) : m_data(new WifiInformationData)
+{
+    set_json(jsonObj);
+}
+
+WifiInformation::WifiInformation(const double signal)
+    : m_data(new WifiInformationData)
+{
+    set_signal(signal);
+}
+
+WifiInformation::WifiInformation(const double signal, const QString ssid, const QString bssid)
+    : m_data(new WifiInformationData)
+{
+    set_signal(signal);
+    set_ssid(ssid);
+    set_bssid(bssid);
+}
+
+WifiInformation::WifiInformation(const WifiInformation &other)
+    : m_data(other.m_data)
+{
+}
+
+WifiInformation &WifiInformation::operator=(const WifiInformation &rhs)
+{
+    if (this != &rhs)
+        m_data.operator=(rhs.m_data);
+    return *this;
+}
+
+WifiInformation::~WifiInformation()
+{
+    m_data.reset();
+}
+
+void WifiInformation::set_ssid(const QString ssid)
```

Comment:
I am getting a lot of Clang-Tidy warnings (not just here but as an example):

![image](https://github.com/ccoors/Valeronoi/assets/8963459/da1d6c7c-3a74-489f-8b87-b792b3714070)

Also, the formatting is quite off. This repo has a few [pre-commit](https://pre-commit.com/) hooks attached, you should also use those. You'll probably need to merge/rebase `main` before they work though.

Reply:
all run an corrected